File: src/main/java/io/github/toolsjs/kubejs/builders/SickleItemBuilder.java
package io.github.toolsjs.kubejs.builders;

import com.blakebr0.cucumber.item.tool.BaseSickleItem;
import dev.latvian.mods.kubejs.item.custom.HandheldItemBuilder;
import dev.latvian.mods.kubejs.typings.Info;
import net.minecraft.resources.ResourceLocation;
import net.minecraft.world.item.Item;

public class SickleItemBuilder extends HandheldItemBuilder {

    private int range = 3;
    public SickleItemBuilder(ResourceLocation i) {
        super(i, 0.5f, -2f);
    }
    @Info(value = "The range of the sickles grass breaking capabilities")
    public SickleItemBuilder range(int range) {
        this.range = range;
        return this;
    }

    @Override
    public Item createObject() {
        return new BaseSickleItem(toolTier,range,properties -> createItemProperties());
    }
}

File: src/main/kotlin/io/github/toolsjs/kubejs/builders/cucumber/SickleItemBuilder.kt
package io.github.toolsjs.kubejs.builders.cucumber

import com.blakebr0.cucumber.item.tool.BaseSickleItem
import dev.latvian.mods.kubejs.item.custom.HandheldItemBuilder
import dev.latvian.mods.kubejs.typings.Info
import net.minecraft.resources.ResourceLocation
import net.minecraft.world.item.Item

class SickleItemBuilder(i: ResourceLocation?) : HandheldItemBuilder(i, 0.5f, -2f) {
    private var breakingRange = 3
    @Info(value = "The range of the sickles grass breaking capabilities")
    fun breakingRange(breakingRange: Int): SickleItemBuilder {
        this.breakingRange = breakingRange
        return this
    }

    override fun createObject(): Item {
        return BaseSickleItem(toolTier, breakingRange) { properties: Item.Properties? -> createItemProperties() }
    }
}